### PR TITLE
Port changes to the number() routine in the micropython library to _number()

### PR DESCRIPTION
**Pull request by hybotix**
This is a port of the changes I did to the number() routine in the micropython version of this library. I also found several errors in processing floats and negative numbers, which I fixed. This version is much more robust. I also fixed a problem with the "from adafruit_ht16k33 import..." line.

### CODE REVIEW

**Comment by hybotix** [comment]
PyLint does not like me.

**Comment by hybotix** [comment]
Done.

File: adafruit_ht16k33/segments.py [hunk 1]
```diff
@@ -26,7 +26,8 @@
 """
 
 from time import sleep
-from adafruit_ht16k33.ht16k33 import HT16K33
+#from adafruit_ht16k33.ht16k33 import HT16K33
+from adafruit_ht16k33.matrix import HT16K33
```
Comment (makermelissa) [inline adafruit_ht16k33/segments.py]: Why was this changed from the base class to the 8x8 matrix class?

File: adafruit_ht16k33/segments.py [hunk 2]
```diff
@@ -211,21 +215,62 @@ def _text(self, text):
...
+            raise ValueError("Input overflow - {0} is too large for the display!".format(number))
+
+        if dot < 0:
+			#	No decimal point (Integer)
+            places = len(stnum)
+        else:
+            places = len(stnum[:dot])
+
+        if self.debug:
+            print("(1) number = {0}, places = {1}, decimal = {2}, dot = {3}, stnum = '{4}'".format(number, places, decimal, dot, stnum))
```
Comment (makermelissa) [inline adafruit_ht16k33/segments.py]: I'd suggest removing the debug statements. They're long and make the library larger. They also seem to be the source of a lot of the linting problems.

File: adafruit_ht16k33/segments.py [hunk 3]
```diff
@@ -211,21 +215,62 @@ def _text(self, text):
...
+    '''
+		Display a floating point or integer number on the Adafruit HT16K33 based displays
+		
+		Param: number - The floating point or integer number to be displayed, which must be
+			in the range 0 (zero) to 9999 for integers and floating point or integer numbers
+			and between 0.0 and 999.0 or 99.00 or 9.000 for floating point numbers.
+		Param: decimal - The number of decimal places for a floating point number if decimal
+			is greater than zero, or the input number is an integer if decimal is zero.
+
+        Returns: The output text string to be displayed.'''
```
Comment (makermelissa) [inline adafruit_ht16k33/segments.py]: This docstring should be moved into the correct function.

File: adafruit_ht16k33/segments.py [hunk 4]
```diff
@@ -211,21 +215,62 @@ def _text(self, text):
...
+            places = 4
+
+            if '.' in stnum:
+                places += 1
+
+        if self.debug:
+            print("(2) places = {0}, dot = {1}, decimal = {2}, stnum = '{3}'".format(places, dot, decimal, stnum))
+
+        #	Set decimal places, if number of decimal places is specified (decimal > 0)	
+        if ((places > 0) and (decimal > 0) and (dot > 0) and (len(stnum[places:]) > decimal)):
```
Comment (makermelissa) [inline adafruit_ht16k33/segments.py]: This is failing pylint because of all the parenthesis and `variable > 0` can be shortened to `variable`. I'd suggest changing this line to:
`if places and decimal and dot and len(stnum[places:]) > decimal:`

File: adafruit_ht16k33/segments.py [hunk 5]
```diff
@@ -211,21 +215,62 @@ def _text(self, text):
...
+        if dot < 0:
+			#	No decimal point (Integer)
+            places = len(stnum)
+        else:
+            places = len(stnum[:dot])
+
+        if self.debug:
+            print("(1) number = {0}, places = {1}, decimal = {2}, dot = {3}, stnum = '{4}'".format(number, places, decimal, dot, stnum))
+
+        if ((places <= 0) and (decimal > 0)):
```
Comment (makermelissa) [inline adafruit_ht16k33/segments.py]: This is failing pylint because of the parenthesis and `variable > 0` can be shortened to `variable`. I'd suggest changing this line to:
`if places <= 0 and decimal:`

**Review by makermelissa — CHANGES_REQUESTED** [review]
Thanks for submitting this. I've made some suggested changes that should help with it passing linting. I'm not sure why you changed it to import from matrix which imports from the base class (ht16k33) rather than just importing the ht16k33 class for better memory usage.

**Comment by hybotix** [comment]
Hi,

I have read your comments and I have a question.

if a and b:
  print("True")

will be True if a is positive and b is negative or vice versa. This would
introduce a bug if for some reason one of them is negative and you *really*
want to check for both > 0. This is why I explicitly check for > 0. I can
drop the parentheses and everything will still come still be right.

What do you think?

File: adafruit_ht16k33/segments.py (hunk 1 above)
Comment (hybotix) [reply adafruit_ht16k33/segments.py]: It makes no sense to me now either. Reversed

File: adafruit_ht16k33/segments.py (hunk 2 above)
Comment (hybotix) [reply adafruit_ht16k33/segments.py]: Removed

File: adafruit_ht16k33/segments.py (hunk 3 above)
Comment (hybotix) [reply adafruit_ht16k33/segments.py]: Done

File: adafruit_ht16k33/segments.py (hunk 4 above)
Comment (hybotix) [reply adafruit_ht16k33/segments.py]: [code]if places and decimal and dot will be true as long as one of them is not zero. This is true for Python 3.7.3. I should be able to remove the parentheses without changing the meaning.

File: adafruit_ht16k33/segments.py (hunk 5 above)
Comment (hybotix) [reply adafruit_ht16k33/segments.py]: Same comment on zero checks.

**Comment by makermelissa** [comment]
Ok yeah, that makes sense. If pylint fails on that, you can add lines to disable certain checks with `#pylint: disable-msg=[pylint-fail-code]` and replace `[pylint-fail-code]` with the actual fail code.

**Comment by hybotix** [comment]
Done! I will get better as I learn what PyLint expects.

File: adafruit_ht16k33/segments.py [hunk 6]
```diff
@@ -26,6 +26,7 @@
 """
 
 from time import sleep
+#from adafruit_ht16k33.ht16k33 import HT16K33
```
Comment (makermelissa) [inline adafruit_ht16k33/segments.py]: Just a minor nitpick, can we remove this commented out line please?

**Review by makermelissa — CHANGES_REQUESTED** [review]
Just one more small thing with regards to some commented out duplicate code, but everything else looks good.

File: adafruit_ht16k33/segments.py (hunk 6 above)
Comment (hybotix) [reply adafruit_ht16k33/segments.py]: Done! I would have nitpicked that too.

**Review by makermelissa — APPROVED** [review]
Looks great! Thanks
